src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocleadingasteriskalign/Example3.java

```diff
@@ -0,0 +1,45 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="JavadocLeadingAsteriskAlign">
+      <property name="tabWidth" value="2"/>
+    </module>
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocleadingasteriskalign;
+
+// xdoc section -- start
+/**
+ * Example with `tabWidth` property.
+ * This example contains Tabs as well as Spaces.
+ */
+public class Example3 {
+  /** <- Preceded with Tabs.
+   * <- Preceded with Tabs & Spaces.
+   */ // <- Preceded with Tabs.
```

Comment:
Are you sure this file has tabs? It is not as drastically crazy spaced as you other input files.

Reply:
This example file does not have Tabs but the test input files has. I did not uses tabs in the example file because it thought that would pointless as examples with tabs and spaces looks same. The violations are correct it just does not use Tabs. Here is the input file which contains tabs and ensures that the check works fine with them: https://github.com/checkstyle/checkstyle/blob/d0b95fd23327fb3b1095da22b976beb983478417/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocleadingasteriskalign/InputJavadocLeadingAsteriskAlignTabs.java

Reply:
This makes this example kind of weird then. One would expect the examples to be what they claim to be. This is why we converted these to compilable files when they were raw text files before.

Even if there is no visible change, I think we should make them tabs. I will create an issue to discuss making tabs more visible somehow.

Reply:
I tried using tabs in the Example file but I was getting this error:

```
[ERROR] [checkstyle] [ERROR] C:\Open Source\checkstyle\src\xdocs\checks\javadoc\javadocleadingasteriskalign.xml:171:1: File contains tab characters (this is the first instance). [FileTabCharacter]
```

I tried adding the suppression in `checkstyle-resources-suppressions.xml`:

```xml
  <suppress checks="FileTabCharacter"
    files="src[\\/]xdocs[\\/]checks[\\/]javadoc[\\/]javadocleadingasteriskalign.xml"/>
````

but it is still giving the same error, I tried changing the regex too like:


```xml
  <suppress checks="FileTabCharacter"
    files="[\\/]javadocleadingasteriskalign.xml"/>
````


```xml
  <suppress checks="FileTabCharacter"
    files="src[\\/]xdocs[\\/]checks[\\/]javadoc[\\/]javadocleadingasteriskalign\.xml"/>
````


```xml
  <suppress checks="FileTabCharacter"
    files="src[\\/]xdocs[\\/]checks[\\/]javadoc[\\/]javadocleadingasteriskalign.*"/>
````

but it is still giving the same error, the path looks valid to me idk why this is happening.

And after doing some research, I found that [FileTabCharacter](https://checkstyle.sourceforge.io/checks/whitespace/filetabcharacter.html)'s examples are also using tabs, some of them are:

[Example1.java](https://github.com/checkstyle/checkstyle/blob/master/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example1.java), [Example2.java](https://github.com/checkstyle/checkstyle/blob/master/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/filetabcharacter/Example2.java), etc... Its all examples are using tabs.

But those tabs are converted into spaces here:

https://github.com/checkstyle/checkstyle/blob/e5c995ea2e6a2db6bf4e6cba2b98f4b26e49b727/src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java#L96-L99

So should we still go with adding tabs in Examples? If yes, then can you please guide me on how to solve the above error? @rnveach 

And sorry for the delay, I was on a little vacation.

Reply:
Are you sure you are adding the suppression to the right place?
https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle-examples-suppressions.xml

Please ensure you are looking at the correct config from the output of the CS run.

Reply:
Added the suppression in your suggested file:

https://github.com/checkstyle/checkstyle/blob/576ca0df85341a61443658f0dacaa4155042292c/config/checkstyle-examples-suppressions.xml#L34-L37

I have added both files, the `Example3` which contains Tabs and `javadocleadingasteriskalign.xml` both. But it still giving the same error

Reply:
Please take a look at the `Tabs` example here:

https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/576ca0d_2024222212/checks/javadoc/javadocleadingasteriskalign.html#

The indentation looks weird because of tabs, I guess we should use the whitespaces only

Reply:
You didn't link to a specific example, but I assume we are talking about https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/576ca0d_2024222212/checks/javadoc/javadocleadingasteriskalign.html#Example3-code .

I am generally fine with this since you specify a tabWidth of 2 but GH and the page display is obviously using something else. It looks like the page is using 8.

If you want to find a work around, you can specify tabWidth as 8, and update your examples accordingly.

Reply:
> Edit: I've kept failing CIs on purpose, to show you where I used the suppressions. 

````
[INFO]      [echo] Checkstyle started (checkstyle-checks.xml): 05/05/2024 10:46:09 PM
[INFO] [checkstyle] Running Checkstyle  on 2133 files
[ERROR] [checkstyle] [ERROR]
````

This config's suppression file is `config/suppressions.xml`.

Reply:
> This config's suppression file is `config/suppressions.xml`.

Thanks now the `mvn clean verify` is passing on local build.

> Every config has it's own suppression file. You will need to identify which config is being run to know which suppression file to modify.

Yeah I just didn't knew which suppression file to use. Thanks for the help

Reply:
> You didn't link to a specific example, but I assume we are talking about https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/576ca0d_2024222212/checks/javadoc/javadocleadingasteriskalign.html#Example3-code .

Yeah I was talking about the Example 3, sorry I forgot to add the link of it

> I am generally fine with this since you specify a tabWidth of 2 but GH and the page display is obviously using something else. It looks like the page is using 8.

I also think they're using tabWidth of 8 the default value.

> If you want to find a work around, you can specify tabWidth as 8, and update your examples accordingly.

If we keep this tabWidth... There will be some misguiding in the example. First the indentation will look weird and second is that, that example will not follow the rule of [CommentIndentation](https://checkstyle.org/checks/misc/commentsindentation.html) module. I'm not sure if that module verifies Javadoc comments or not but I guess it'll lead user to think that it is okay to place comments at any indentation level and it'll not cause any error or something like that. I guess we should keep using the whitespaces after all, the example will not contain any weird indentations and will not misguide the user. 

I'll push the changes with tabs removed in the Example3.java